app.ts

```diff
@@ -0,0 +1,252 @@
+				$("#tableBody").empty();
+				for (const record of records) {
+					// creates row for each record
+					$("tbody").append(`<tr class="row"></tr>`);
+					const lastRow = $(".row:last");
+					for (const value of record) {
+						// assign each record to their column in a specified row
+						lastRow.append(`<td>${value}</td>`);
+						if (value === inputValue) {
+							// highlights the searched row 
+							lastRow.css('background-color', '#DDC0B4');
+						}
+					}
+					$("tbody").append(lastRow);
+				}
+				$('#page').empty().append(`Showing record: ${this.firstNumber} - ${this.lastNumber}`);
+				$('#loader').hide();
+			})
+			.catch(err => {
+				alert('Error while displaying records, reload the page');
+				throw err;
+			});
+	}
+
+	/** recalculates the record range that includes inputValue fromm user */
+	searchRecordsAndResize() {
+		let inputValue = <number>($('#searchInput').val());
+		if (inputValue < 0 || inputValue > this.recordCount) {
+			$('.modal').css('display', 'block');
+			$('.content').append(`<p>${inputValue} is not a number within the range. Please try a different number</p>`);
+			$('#page').empty().append(`Showing record: ${this.firstNumber} - ${this.lastNumber}`);
+			$('#searchInput').val('');
+			return;
+		}
+		let calculatedRows = this.calculatingRows();
+		// divides the calculated max rows in half
+		const halfRange = Math.floor(calculatedRows / 2);
+		this.firstNumber = Math.max(0, inputValue - halfRange);
+		this.lastNumber = Math.min(this.recordCount, this.firstNumber + (calculatedRows - 1));
+	}
+
+	/** Navigates to the next set of records */
+	rightArrow() {
+		$('#searchInput').val('');
+		// retrieves the last row
+		const lastRow = document.querySelector("#recordsTable tbody .row:last-child");
+		// checks if the last row exists 
+		if (lastRow) {
+			const cells = lastRow.querySelectorAll("td");
+			if (cells.length > 0) {
+				// Get the value of the first cell 
+				const lastID = parseFloat(cells[0].textContent || "");
+				// checks if the last value is within range 
+				if (0 <= lastID && lastID <= this.recordCount) {
+					// calculates the first number of the page 
+					this.firstNumber = lastID + 1;
+					const calculatedRows = this.calculatingRows();
+					// calculates the last number of the page 
+					this.lastNumber = this.firstNumber + (calculatedRows - 1);
+				}
+			}
+		}
+		this.updateAndDisplayRecords();
+	}
+
+	leftArrow() {
+		$('#searchInput').val('');
+		// retrieves the first row 
+		const firstRow = document.querySelector("#recordsTable tbody .row:first-child");
+		if (firstRow) {
+			const cells = firstRow.querySelectorAll("td");
+			if (cells.length > 0) {
+				const firstID = parseFloat(cells[0].textContent || "");
+				if (0 <= firstID && firstID <= this.recordCount) {
+					const calculatedRows = this.calculatingRows();
+					this.lastNumber = firstID - 1;
+					this.firstNumber = this.lastNumber - (calculatedRows - 1);
+				}
```

Comment:
Similar comment to the `rightArrow` function's comment. You don't have to rely on the HTML. You are the point of truth. Also, maybe a more descriptive function name instead of rightArrow and leftArrow? I mean, what does the function do? Navigate back and forth. Verbs instead.